internal/stackdiag.go

```diff
@@ -379,44 +383,95 @@ func scheduleJobs(k *Kubectl, ns string, recordErr func(error), state *diagJobSt
+		if esName != "" {
+			logger.Print("scheduleJob", "typ ", typ, "es ", esName)
+			recordErr(state.scheduleJob(typ, esName, ressourceInfo.Name, isTLS))
+		}
+		return nil
+	})
+}
+
+func extractEsInfo(typ string, ns string, resourceInfo *resource.Info, filters internal_filters.Filters) (bool, string, error) {
+	resourceName := resourceInfo.Name
+
+	if !filters.Empty() && !filters.Contains(resourceName, typ) {
+		return false, "", nil
+	}
+
+	es, err := runtime.DefaultUnstructuredConverter.ToUnstructured(resourceInfo.Object)
+	if err != nil {
+		return false, "", err
+	}
+
+	var isTLS bool
+	switch typ {
+	case logstashJob:
+		// Logstash API SSL is not yet configurable via spec.http.tls, try to read the config as a best-effort
+		// To change after https://github.com/elastic/cloud-on-k8s/issues/6971 is fixed.
+		enabled, found, err := unstructured.NestedBool(es, "spec", "config", "api.ssl.enabled")
 		if err != nil {
-			recordErr(err)
-			return nil
+			return false, "", err
 		}
-		tls := !(found && disabled)
+		isTLS = found && enabled
 
-		if !filters.Empty() && !filters.Contains(resourceName, typ) {
-			return nil
+	default:
+		disabled, found, err := unstructured.NestedBool(es, "spec", "http", "tls", "selfSignedCertificate", "disabled")
+		if err != nil {
+			return false, "", err
 		}
+		isTLS = !(found && disabled)
+	}
 
-		esName := resourceName
-		if typ != "elasticsearch" {
-			val, found, err := unstructured.NestedString(es, "spec", "elasticsearchRef", "name")
-			if err != nil {
-				recordErr(err)
-				return nil
-			}
-			if !found || val == "" {
-				logger.Printf("Skipping %s/%s as it it not using elasticsearchRef", ns, resourceName)
-				return nil
-			}
-			esName = val
+	var esName string
+	switch typ {
+	case elasticsearchJob:
+		esName = resourceName
+	case kibanaJob:
+		name, found, err := unstructured.NestedString(es, "spec", "elasticsearchRef", "name")
+		if err != nil {
+			return false, "", err
 		}
+		if !found || name == "" {
+			logger.Printf("Skipping %s/%s as elasticsearchRef is not defined", ns, resourceName)
+			return false, "", nil
+		}
+		esName = name
+	case logstashJob:
+		esRefs, found, err := unstructured.NestedSlice(es, "spec", "elasticsearchRefs")
+		if err != nil {
+			return false, "", err
+		}
+		if !found || len(esRefs) == 0 {
+			logger.Printf("Skipping %s/%s as elasticsearchRefs is not defined", ns, resourceName)
+			return false, "", nil
+		}
+		esRef, ok := esRefs[0].(map[string]interface{})
+		if !ok {
+			logger.Printf("Skipping %s/%s as elasticsearchRefs[0] is invalid", ns, resourceName)
+			return false, "", nil
+		}
+		name, found, err := unstructured.NestedString(esRef, "name")
+		if err != nil {
+			return false, "", err
+		}
+		if !found || name == "" {
+			logger.Printf("Skipping %s/%s as name is not set in elasticsearchRefs[0]", ns, resourceName)
+			return false, "", nil
+		}
+		esName = name
```

Comment:
I am wrong here. The Logstash API doesn't have authentication configured by default and when it does, it uses its own credentials, not the ES credentials.

Reply:
Changed in 1c2c77f.